crates/cli/src/main.rs

```diff
@@ -63,10 +65,14 @@ fn main() -> Result<()> {
                 .source_compression(codegen.source_compression)
                 .provider_version("2");
 
+            let js_runtime_options: JsRuntimeOptionGroup = opts.js_runtime.clone().into();
             let mut gen = if codegen.dynamic {
-                builder.build::<DynamicGenerator>()?
+                if js_runtime_options != JsRuntimeOptionGroup::default() {
+                    bail!("Cannot set JS runtime options when building a dynamic module");
+                }
+                builder.build_dynamic()?
```

Comment:
On a second thought, I wonder if the builder is better suited to perform this validation? 

That way:

1. We can keep the current `build` method, shared across dynamic/static codegen.
2. Perform the validation of the option groups through `javy::Config`
3. Keep main as slim as possible i.e., avoid baking in builder specific logic.

Reply:
In other words, we could modify the existing build method signature to be:

`fn build<T>(config: javy_config::Config) -> Result<..>`

Reply:
At some point in the future, if the defaults for `build` and `compile` diverge, the logic for validating the config for dynamic is going to depend on whether the build or compile command was used. But we can deal with that case when it actually comes up I guess.